enhancements/installer/compact-plus-worker.md

```diff
@@ -0,0 +1,188 @@
+---
+title: Compact + Worker deployment as day0 operation
+authors:
+  - "@flaper87"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+api-approvers: # in case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers)
+  - TBD
+creation-date: 2022-03-08
+last-updated: 2022-03-08
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Compact + Worker deployment as day0 operation
+
+## Summary
+
+This enhancement proposes to allow for deploying a 4-nodes cluster as a day 0
+operation. The 4-nodes cluster would be composed by 3 schedulable master nodes
+and a single worker, which provide users with a highly available cluster.
+
+## Motivation
+
+There are environments where only 4 nodes are available where 4-nodes clusters
+are being deployed. We could serve OpenShift cluster administrators of these
+environments better by allowing them to deploy a 4-nodes cluster as a day0
+operation.
+
+Today, in such environments, OpenShift cluster administrators can only deploy a
+4-nodes clusters by first deploying a compact cluster (3 schedulable masters)
+and then adding a host as a day 2 operation. This provides the same results as
+aimed in this enhancement but with a poor and slower user experience.
+
+Today, because of the current limitations in the installer, it takes
+administrators significantly longer to achieve the desired topology, when only
+4 nodes are available. Furthermore, by requiring more steps, we are making the
+process more prone to user errors.
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+
+- Enable the installer to deploy a 4-nodes cluster composed by 3 schedulable
+  masters and one worker.
+
+### Non-Goals
+
+- Define new infrastructure topologies
+
+- Define new architectures
+
+- Provide official support for the 4-nodes cluster deployments
+
+## Proposal
+
+- Set the InfrastructureTopology to have the same value as the
+  ControlPlaneTopology when the number of workers is 1, instead of setting its
+  value to SingleReplica as it's done today.
```

Comment:
```suggestion
- Set the `InfrastructureTopology` to have the same value as the
  `ControlPlaneTopology` when the number of workers is 1, instead of setting its
  value to `SingleReplica` as it's done today.
```